docker/models/containers.py

```diff
@@ -709,7 +716,14 @@ def run(self, image, command=None, stdout=True, stderr=False,
         if exit_status != 0:
             stdout = False
             stderr = True
-        out = container.logs(stdout=stdout, stderr=stderr)
+
+        logging_driver = container.attrs['HostConfig']['LogConfig']['Type']
+
+        if logging_driver == 'json-file' or logging_driver == 'journald':
```

Comment:
Question is if different logic is needed based on the error; running with docker directly;

```
$ docker run -dit --log-driver=none --name foo busybox

$ docker logs foo
Error response from daemon: configured logging driver does not support reading
```